File: src/main/java/com/methodpark/cmpe/code/dip/ServerLogFileParser.java
package com.methodpark.cmpe.code.dip;

import java.util.ArrayList;
import java.util.List;

public class ServerLogFileParser
{
    private ApacheLog apacheLog;

    public ServerLogFileParser(ApacheLog apacheLog)
    {
        this.apacheLog = apacheLog;
    }

    public List<String> getErrors()
    {
        List<String> errors = new ArrayList<String>();
        List<LogEntry> logEntries = apacheLog.loadLogEntries();
        for (LogEntry logEntry : logEntries)
        {
            if (logEntry.getLogType() == LogType.Error)
            {
                errors.add(logEntry.getMessage());
            }
        }
        return errors;
    }
}

File: src/main/csharp/dip/Dip/ServerLogFileParser.cs
using System.Collections.Generic;

namespace Dip
{
    public class ServerLogFileParser
    {
        private readonly ApacheLog apacheLog;

        public ServerLogFileParser(ApacheLog apacheLog)
        {
            this.apacheLog = apacheLog;
        }

        public List<string> GetErrors()
        {
            var errors = new List<string>();
            foreach (var logEntry in apacheLog.LoadLogEntries())
            {
                if (logEntry.Type == LogType.Error)
                {
                    errors.Add(logEntry.Message);
                }
            }
            return errors;
        }
    }
}